File: user-service/src/main/java/com/athena/userservice/model/AdminUser.java
package com.athena.userservice.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "admin_users")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AdminUser {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false)
    private String username;

    @Column(name = "password_hash", nullable = false)
    private String passwordHash;

    @Column(name = "first_name")
    private String firstName;

    @Column(name = "last_name")
    private String lastName;

    @Column
    private String email;

    /** Role: ADMIN | ANALYST | VIEWER | CREDIT_RISK */
    @Column(nullable = false)
    private String role;

    @Column(name = "totp_secret")
    private String totpSecret;

    @Column(name = "is_active", nullable = false)
    @Builder.Default
    private boolean active = true;
}

File: scoring-service/internal/model/admin_user.go
package model

type AdminUser struct {
	ID           int64  `gorm:"column:id;primaryKey;autoIncrement" json:"id"`
	Username     string `gorm:"column:username;uniqueIndex;not null" json:"username"`
	PasswordHash string `gorm:"column:password_hash;not null" json:"-"`
	FirstName    string `gorm:"column:first_name" json:"first_name"`
	LastName     string `gorm:"column:last_name" json:"last_name"`
	Email        string `gorm:"column:email" json:"email"`
	Role         string `gorm:"column:role;not null" json:"role"`
	TotpSecret   string `gorm:"column:totp_secret" json:"-"`
	Active       bool   `gorm:"column:is_active;not null;default:true" json:"active"`
}

func (AdminUser) TableName() string {
	return "admin_users"
}
